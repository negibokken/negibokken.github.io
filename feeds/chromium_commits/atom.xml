<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <id>https://negibokken.github.io/feeds/chromium_commits/atom.xml</id>
    <title>intent to ship feed</title>
    <link href="https://negibokken.github.io/feeds/chromium_commits/atom.xml" rel="self"/>
    <updated>2023-04-15T00:14:11.000Z</updated>
</feed>