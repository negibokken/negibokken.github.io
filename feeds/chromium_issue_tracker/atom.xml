<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
    <id>https://negibokken.github.io/feeds/chromium_issue_tracker/atom.xml</id>
    <title>Chromium issue tracker feed</title>
    <link href="https://negibokken.github.io/feeds/intent_to_ship/atom.xml" rel="self"/>
    <updated>Oct 8, 2024 04:17AM</updated>
    <entry>
        <id>372093897</id>
        <title>【Chromium &amp;gt; Speed (Use Subcomponents) &amp;gt; Regressions】 [Graphics]: [1] regressions in rendering.desktop</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372093897"/>
        <updated>Oct 8, 2024 04:17AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372044255</id>
        <title>【Chromium &amp;gt; Mobile &amp;gt; WebView】 Error closing apps</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372044255"/>
        <updated>Oct 8, 2024 03:51AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372119113</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; Input】 Tests are failing: ninja://android_webview/test:trichrome_webview_wpt_64/external/wpt/visual-viewport/scroll-event-order.html</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372119113"/>
        <updated>Oct 8, 2024 02:55AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372119276</id>
        <title>【Chromium &amp;gt; Platform &amp;gt; WebAppProvider】 Step "browser_tests (retry shards) on Mac-13" failing on builder "chromium/ci/Mac13 Tests"</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372119276"/>
        <updated>Oct 8, 2024 02:52AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062071</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; JavaScript &amp;gt; Compiler】 Dwitter demo (https://beta.dwitter.net/d/9968) is 2x slower than Firefox (time spent around JS)</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062071"/>
        <updated>Oct 8, 2024 02:46AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062070</id>
        <title>【Chromium】 don't allow print preview for non-tab contexts</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062070"/>
        <updated>Oct 8, 2024 02:18AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372064004</id>
        <title>【Chromium &amp;gt; Blink】 Video not rendered when displaying multiple &amp;lt;video&amp;gt; elements with same mp4 as src</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372064004"/>
        <updated>Oct 8, 2024 01:48AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372053395</id>
        <title>【Chromium &amp;gt; Blink】 react is not working</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372053395"/>
        <updated>Oct 8, 2024 01:12AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372044252</id>
        <title>【Chromium &amp;gt; Blink】 CSS transform with perspective and/or rotateX breaks z-index for the rest of the document</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372044252"/>
        <updated>Oct 8, 2024 12:34AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097716</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; TopChrome &amp;gt; TabStrip】 Declutter UI not updating on the timed stale tabs recalculation</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097716"/>
        <updated>Oct 8, 2024 12:29AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097609</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; TopChrome &amp;gt; TabStrip】 Disable declutter in incognito mode</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097609"/>
        <updated>Oct 8, 2024 12:28AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097785</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; TopChrome &amp;gt; TabStrip】 Close the tab search bubble on "Close tabs" click</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097785"/>
        <updated>Oct 8, 2024 12:28AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097697</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; TopChrome &amp;gt; TabStrip】 Update the stale tabs list on visibilitychanged</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097697"/>
        <updated>Oct 8, 2024 12:27AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372098338</id>
        <title>【Chromium &amp;gt; Internals &amp;gt; Compositing】 compositor_frame_fuzzer: Abrt in logging::CheckLogMessage::~CheckLogMessage</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372098338"/>
        <updated>Oct 8, 2024 12:27AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097110</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; TopChrome &amp;gt; TabStrip】 Use an enum rather than an int for the tab search tab index, for readability</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097110"/>
        <updated>Oct 8, 2024 12:25AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372097205</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; Input】 Tests are failing: ninja://android_webview/test:trichrome_webview_wpt_64/external/wpt/visual-viewport/page-and-offset-in-iframe.html</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372097205"/>
        <updated>Oct 8, 2024 12:25AM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059341</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; MediaRecording】 Edge FeedbackHub crash on the renderer during recording.</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059341"/>
        <updated>Oct 7, 2024 11:50PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059340</id>
        <title>【Chromium &amp;gt; Build】 d3dcompiler_47.dll does not load under gl_tests.exe on Windows 1909 importing __CxxFrameHandler4</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059340"/>
        <updated>Oct 7, 2024 11:35PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372086602</id>
        <title>【Chromium &amp;gt; Internals &amp;gt; Updater】 Investigate use-after-move clang-tidy miss in GoogleUpdater</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372086602"/>
        <updated>Oct 7, 2024 11:23PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062068</id>
        <title>【Chromium &amp;gt; Platform &amp;gt; WebAppProvider &amp;gt; IntegrationTesting】 Improve checks in WebAppSyncBridgeTest related to source/management type</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062068"/>
        <updated>Oct 7, 2024 11:18PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372064003</id>
        <title>【Chromium】 My Chromebook browser crashes when searching up 'auto' in settings</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372064003"/>
        <updated>Oct 7, 2024 11:02PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372064002</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; WebUI】 FlagsUI: Changing a &amp;lt;select&amp;gt;'s value twice and blurring results in a runtime assertion error.</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372064002"/>
        <updated>Oct 7, 2024 10:55PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372064001</id>
        <title>【Chromium &amp;gt; Services (Use Subcomponents) &amp;gt; Safebrowsing】 Implement telemetry for redirection to external apps</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372064001"/>
        <updated>Oct 7, 2024 10:50PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372080121</id>
        <title>【Chromium &amp;gt; Dawn &amp;gt; Bindings &amp;gt; Emscripten】 Implement WGPUFuture version of getCompilationInfo in Emscripten</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372080121"/>
        <updated>Oct 7, 2024 10:41PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372080570</id>
        <title>【Chromium &amp;gt; Dawn &amp;gt; Bindings &amp;gt; Emscripten】 Implement WGPUFuture version of popErrorScope in Emscripten</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372080570"/>
        <updated>Oct 7, 2024 10:41PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372079137</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; Shopping】 Use UIViewContentModeScaleAspectFill not UIViewContentModeScaleAspectFit</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372079137"/>
        <updated>Oct 7, 2024 10:37PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372077545</id>
        <title>【Chromium &amp;gt; Dawn &amp;gt; Bindings &amp;gt; Emscripten】 Implement WGPUFuture version of onSubmittedWorkDone in Emscripten</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372077545"/>
        <updated>Oct 7, 2024 10:24PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372064000</id>
        <title>【Chromium &amp;gt; Platform &amp;gt; Apps &amp;gt; Container &amp;gt; Fullscreen】 Black screen when opening a window from a fullscreen PWA.</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372064000"/>
        <updated>Oct 7, 2024 10:24PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062067</id>
        <title>【Chromium &amp;gt; Build &amp;gt; Rust】 The CXX bridge fails to parse bindgen-related snippets</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062067"/>
        <updated>Oct 7, 2024 10:07PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372044249</id>
        <title>【Chromium &amp;gt; Dawn】 Compat CTS: textureStore:out_of_bounds_array tests failing on Intel 770 Linux GL</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372044249"/>
        <updated>Oct 7, 2024 10:02PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372044248</id>
        <title>【Chromium &amp;gt; Dawn】 Compat CTS: limits tests failing on Linux NVidia/GL</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372044248"/>
        <updated>Oct 7, 2024 09:54PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062064</id>
        <title>【Chromium &amp;gt; PDFium】 Add FPDFPageObjMark_{Get,Set}FloatParam() APIs</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062064"/>
        <updated>Oct 7, 2024 09:43PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372066818</id>
        <title>【Chromium &amp;gt; UI &amp;gt; Browser &amp;gt; UserEducation】 Updated BrowserUserEducationInterface API</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372066818"/>
        <updated>Oct 7, 2024 09:42PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062063</id>
        <title>【Chromium &amp;gt; PDFium】 Unify FPDFPageObjMark_{Get,Set}*Param() implementations</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062063"/>
        <updated>Oct 7, 2024 09:41PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059337</id>
        <title>【Chromium】 codes and license</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059337"/>
        <updated>Oct 7, 2024 09:31PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372044247</id>
        <title>【Chromium】 Startup on Fedora 40 takes 13 seconds</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372044247"/>
        <updated>Oct 7, 2024 09:21PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372062982</id>
        <title>【Chromium &amp;gt; Internals &amp;gt; Views】 `BrowserNonClientFrameViewChromeOSTest.BrowserHeaderVisibilityInTabletModeTest` fails: `browser_view-&amp;gt;needs_layout()` unexpectedly `true`</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372062982"/>
        <updated>Oct 7, 2024 09:20PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372063998</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; HTML &amp;gt; Modules】 Create Issue Test</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372063998"/>
        <updated>Oct 7, 2024 09:20PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372053392</id>
        <title>【Chromium &amp;gt; Internals &amp;gt; Preload】 Implement a viewport-based preloading predictor for Mobile</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372053392"/>
        <updated>Oct 7, 2024 09:12PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059336</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; JavaScript】 roll test262 failures on built-ins/String and built-ins/RegExp tests</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059336"/>
        <updated>Oct 7, 2024 09:08PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059335</id>
        <title>【Chromium &amp;gt; UI】 I Want Chromium license code and UI for windows and iOS devices media audio videos</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059335"/>
        <updated>Oct 7, 2024 09:04PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371949939</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; JavaScript】 roll test262 failures on language/expressions tests</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371949939"/>
        <updated>Oct 7, 2024 09:02PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372059334</id>
        <title>【Chromium &amp;gt; Platform &amp;gt; WebAppProvider】 Add test infrastructure to support redirection and waiting for correct message</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372059334"/>
        <updated>Oct 7, 2024 08:56PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371949938</id>
        <title>【Chromium &amp;gt; Platform &amp;gt; DevTools】 Chrome doesn't load preloaded 103 Early Hints resources from disk cache</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371949938"/>
        <updated>Oct 7, 2024 08:42PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372055494</id>
        <title>【Chromium &amp;gt; Internals &amp;gt; Preload &amp;gt; PrefetchProxy】 Make prefetch network request failures look less fatal</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372055494"/>
        <updated>Oct 7, 2024 08:37PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371898615</id>
        <title>【Chromium &amp;gt; Speed (Use Subcomponents) &amp;gt; Tracing】 v8 methodload and scriptload ETW events are don't get captured if JavaScript isn't running</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371898615"/>
        <updated>Oct 7, 2024 08:13PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371971708</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; WebGL】 Context loss + enum WebGL conformance tests failing after WebGL roll</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371971708"/>
        <updated>Oct 7, 2024 07:52PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371898614</id>
        <title>【Chromium &amp;gt; UI】 Not working properly</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371898614"/>
        <updated>Oct 7, 2024 07:45PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>371659899</id>
        <title>【Chromium &amp;gt; Blink &amp;gt; Editing &amp;gt; Selection】 [WPT] New failures introduced in external/wpt/selection by import https://crrev.com/c/5913172</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/371659899"/>
        <updated>Oct 7, 2024 07:43PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
    <entry>
        <id>372041032</id>
        <title>【Chromium &amp;gt; UI】 Chrome mobile bug with YouTube scrolling</title>
        <author>
            <name>Chromium issue tracker</name>
        </author>
        <link href="https://issues.chromium.org/issues/372041032"/>
        <updated>Oct 7, 2024 07:43PM</updated>
        <summary>Click to view details</summary>
        <content type="html">
X���/��z֢�</content>
    </entry>
</feed>