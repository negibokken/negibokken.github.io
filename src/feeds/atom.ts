interface AtomEntryProps {
    id: string;
    title: string;
    author: {
        name: string;
    }
    link: string;
    updated: string;
    summary: string;
    content: string;
}

export class AtomEntry {
    _name: string = 'entry';
    id: string;
    title: string;
    author: {
        name: string;
    };
    updated: string;
    link: {
        _attr: {
            href: string;
        }
    };
    summary: string;
    content: string;
    constructor({ id, title, author, updated, link, summary, content }: AtomEntryProps) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.link = {
            _attr: {
                href: link,
            }
        };
        this.updated = updated;
        this.summary = summary;
        this.content = content;
    }

    toXML(): string {
        return transformToXML(this);
    }
}

interface AtomFeedProps {
    id: string;
    title: string;
    link: string;
    updated: string;
    entries?: AtomEntry[];
}

export class AtomFeed {
    _name: string = 'feed';
    id: string;
    title: string;
    link: {
        _attr: {
            href: string;
            rel: string;
        }
    };
    updated: string;
    entries: AtomEntry[];
    _attr: {
        xmlns: string;
    };
    constructor({ id, title, link, updated, entries = [] }: AtomFeedProps) {
        this.id = id;
        this.title = title;
        this.link = {
            _attr: {
                href: link,
                rel: 'self',
            }
        };
        this.updated = updated;
        this.entries = entries;
        this._attr = {
            xmlns: "http://www.w3.org/2005/Atom"
        };
    }

    appendEntry(entry: AtomEntry): void {
        this.entries.push(entry);
    }

    toXML(): string {
        const template = '<?xml version="1.0" encoding="utf-8"?>';
        return template + transformToXML(this);
    }
}

function transformToXML(obj: any): string {
    let attr = "";
    if (typeof obj === 'object' && '_attr' in obj) {
        attr = " " + Object.entries(obj["_attr"]).map(([_key, _value]) => `${_key}="${_value}"`).join(" ");
    }

    // We should close tag if element doesn't have the inner contents.
    const innerContent = Object.entries(obj).filter(([key, value]) => !key.startsWith("_")).length;
    if (!innerContent) {
        return `<${obj._name}${attr} />`;
    }

    let xml = obj._name ? `<${obj._name}${attr}>` : '';

    for (const [key, value] of Object.entries(obj)) {
        if (key.startsWith("_")) continue;
        if (Array.isArray(value)) {
            xml += transformToXML(value);
        } else if (typeof value === 'object') {
            xml += transformToXML({ _name: key, ...value });
        } else {
            xml += `<${key}>${value}</${key}>`;
        }
    }
    xml += obj._name ? `</${obj._name}>` : '';
    return xml;
}
