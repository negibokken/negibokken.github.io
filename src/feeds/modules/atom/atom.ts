import convert from 'xml-js';
export interface AtomEntryProps {
    id: string;
    title: string;
    author: {
        name: string;
    }
    link: string;
    updated: string;
    summary: string;
    content: string;
}

export class AtomEntry {
    _name: string = 'entry';
    id: string;
    title: string;
    author: {
        name: string;
    };
    updated: string;
    link: {
        _attr: {
            href: string;
        }
    };
    summary: string;
    content: {
        _attr: {
            type: string
        };
        _text: string;
    };
    constructor({ id, title, author, updated, link, summary, content }: AtomEntryProps) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.link = {
            _attr: {
                href: link,
            }
        };
        this.updated = updated;
        this.summary = summary;
        this.content = {
            _attr: {
                type: 'html',
            },
            _text: content
        }
    }

    toXML(): string {
        return transformToXML(this);
    }
}

interface AtomFeedProps {
    id: string;
    title: string;
    link: string;
    updated: string;
    entries?: AtomEntry[];
}

export class AtomFeed {
    _name: string = 'feed';
    id: string;
    title: string;
    link: {
        _attr: {
            href: string;
            rel: string;
        }
    };
    updated: string;
    entries: AtomEntry[];
    _attr: {
        xmlns: string;
    };
    constructor({ id, title, link, updated, entries = [] }: AtomFeedProps) {
        this.id = id;
        this.title = title;
        this.link = {
            _attr: {
                href: link,
                rel: 'self',
            }
        };
        this.updated = updated;
        this.entries = entries;
        this._attr = {
            xmlns: "http://www.w3.org/2005/Atom"
        };
    }

    appendEntry(entry: AtomEntry): void {
        this.entries.push(entry);
    }

    toXML(): string {
        const template = '<?xml version="1.0" encoding="utf-8"?>';
        return template + transformToXML(this);
    }

    static createFromXML(xml: string): AtomFeed {
        try {
            const json = JSON.parse(convert.xml2json(xml, { compact: true, spaces: 4 }));
        } catch (e) {
            console.error(xml);
            throw new Error(`Failed to parse XML: ${e}`);
        }
        console.log(JSON.stringify(json, null, '  '));
        return new AtomFeed({
            id: json.feed.id._text,
            title: json.feed.title._text,
            link: json.feed.link._attributes.href,
            updated: json.feed.updated._text,
            entries: json.feed.entry.map((entry: any) => new AtomEntry({
                id: entry.id._text,
                title: entry.title._text,
                author: { name: entry.author.name._text },
                content: entry.content._text,
                link: entry.link._attributes.href,
                summary: entry.summary._text,
                updated: entry.updated._text,
            }))
        });
    }
}

function transformToXML(obj: any): string {
    let attr = "";
    if (typeof obj === 'object' && '_attr' in obj) {
        attr = " " + Object.entries(obj["_attr"]).map(([_key, _value]) => `${_key}="${_value}"`).join(" ");
    }

    // We should close tag if element doesn't have the inner contents.
    const innerContent = Object.entries(obj).filter(([key, value]) => !key.startsWith("_") || key === "_text").length;
    if (!innerContent) {
        return `<${obj._name}${attr} />`;
    }

    if (typeof obj === 'object' && '_text' in obj) {
        return `<${obj._name}${attr}>${obj._text}</${obj._name}>`;
    }

    let xml = obj._name ? `<${obj._name}${attr}>` : '';

    for (const [key, value] of Object.entries(obj)) {
        if (key.startsWith("_")) continue;
        if (Array.isArray(value)) {
            xml += transformToXML(value);
        } else if (typeof value === 'object') {
            xml += transformToXML({ _name: key, ...value });
        } else {
            xml += `<${key}>${value}</${key}>`;
        }
    }
    xml += obj._name ? `</${obj._name}>` : '';
    return xml;
}
