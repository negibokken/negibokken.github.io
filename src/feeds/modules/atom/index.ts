export { AtomFeed, AtomEntry } from './atom';
