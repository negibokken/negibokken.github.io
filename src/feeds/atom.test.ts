import { AtomEntry, AtomFeed } from './atom';
import { describe, it, expect } from 'vitest';
import { format } from 'prettier';

describe('Atom', () => {
    it('AtomEntry toXML', () => {
        const entry = new AtomEntry({
            id: 'https://example.com/a',
            title: "title",
            author: {
                name: 'author',
            },
            updated: "2001-02-03T16:05:06Z",
            link: "https://example.com",
            summary: "this is a summary text",
            content: "this is a content of the entry",
        });
        console.log(format(entry.toXML(), { parser: 'xml' }));
        expect(format(entry.toXML(), { parser: 'xml' })).toMatchSnapshot();
    });

    it('AtomFeed toXML', () => {
        const entry = new AtomEntry({
            id: 'https://example.com/a',
            title: "title",
            author: {
                name: 'author',
            },
            updated: "2001-02-03T16:05:06Z",
            link: "https://example.com",
            summary: "this is a summary text",
            content: "this is a content of the entry",
        });
        const feed = new AtomFeed({
            id: 'https://example.com',
            title: 'feed_title',
            link: 'https://example.com',
            updated: "2001-02-03T16:05:06Z",
        });
        feed.appendEntry(entry);
        console.log(format(feed.toXML(), { parser: 'xml' }));
        expect(format(feed.toXML(), { parser: 'xml' })).toMatchSnapshot();
    });
});
