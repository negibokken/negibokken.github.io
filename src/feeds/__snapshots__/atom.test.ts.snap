// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Atom > AtomEntry toXML 1`] = `
"<entry><id>https://example.com/a</id><title>title</title><author><name
    >author</name></author><link href=\\"https://example.com\\" /><updated
  >2001-02-03T16:05:06Z</updated><summary
  >this is a summary text</summary><content
    type=\\"html\\"
  >this is a content of the entry</content></entry>
"
`;

exports[`Atom > AtomFeed toXML 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\" ?>
<feed xmlns=\\"http://www.w3.org/2005/Atom\\"><id>https://example.com</id><title
  >feed_title</title><link href=\\"https://example.com\\" rel=\\"self\\" /><updated
  >2001-02-03T16:05:06Z</updated><entry><id>https://example.com/a</id><title
    >title</title><author><name>author</name></author><link
      href=\\"https://example.com\\"
    /><updated>2001-02-03T16:05:06Z</updated><summary
    >this is a summary text</summary><content
      type=\\"html\\"
    >this is a content of the entry</content></entry></feed>
"
`;
